devito/dse/rewriters.py

```diff
@@ -406,6 +408,38 @@ def _extract_sum_of_products(self, cluster, template, **kwargs):
 
         return cluster.rebuild(processed)
 
+    @dse_pass
+    def _skewing(self, cluster, template, **kwargs):
+        """
+        A pass to perform only the basic skewing in order to have valid data
+        dependences.
+        """
+        skew_factor = configuration['skew_factor']
+        skew_dim, mapper, int_mapper = None, {}, []
+        total_int = {}
+        intervals, sub_iterators, directions = cluster.ispace.args
+
+        for dim in cluster.ispace.dimensions:
+            if isinstance(dim, SpaceDimension):
+                mapper[dim] = dim + skew_factor*skew_dim
+                int_mapper.append(IntervalGroup([Interval(dim, -skew_factor*skew_dim,
```

Comment:
You don't need an IntervalGroup at this point.

Also, ``int_mapper`` should actually be called ``intervals``